Lib/test/test_tarfile.py

```diff
@@ -690,6 +691,36 @@ def format_mtime(mtime):
             tar.close()
             os_helper.rmtree(DIR)
 
+    @unittest.skipUnless(hasattr(errno, "EFTYPE"), "errno.EFTYPE required")
+    def test_extract_chmod_eftype(self):
+        # Extracting a file as non-root should skip the sticky bit (gh-108948)
+        # even on platforms where chmod fails with EFTYPE (i.e. FreeBSD). But
+        # we need to take care that any other error is preserved.
+        mode = "-rwxrwxrwt"
+        with ArchiveMaker() as arc:
+            arc.add("sticky1", mode=mode)
+            arc.add("sticky2", mode=mode)
+        tar = arc.open(errorlevel=2)
+        DIR = os.path.join(TEMPDIR, "chmod")
+        os.mkdir(DIR)
+        self.addCleanup(os_helper.rmtree, DIR)
+        with tar:
+            # this should not raise:
+            tar.extract("sticky1", DIR, filter="fully_trusted")
+            got_mode = stat.filemode(os.stat(os.path.join(DIR, "sticky1")).st_mode)
+            expected_mode = "-rwxrwxrwx" if os.geteuid() != 0 else "-rwxrwxrwt"
+            self.assertEqual(got_mode, expected_mode)
+
+            # but we can create a situation where it does raise:
+            with unittest.mock.patch("os.chmod") as mock:
+                eftype_error = OSError(errno.EFTYPE, "EFTYPE")
+                other_error = OSError(errno.EPERM, "different error")
+                mock.side_effect = [eftype_error, other_error]
+                with self.assertRaises(tarfile.ExtractError) as excinfo:
+                    tar.extract("sticky2", DIR, filter="fully_trusted")
```

Comment:
Since tarfile implementation now uses a complicated triple-nested try block, would it be possible to test also the othe code path: EFTYPE error but then the second chmod() succeed.